### ISSUE TITLE
Type Variabels in Flix Doc

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#1146]
We need to update the documentor to print the names of type variables and not their symbols:

![image](https://user-images.githubusercontent.com/2287953/83003547-c008a600-a00e-11ea-8e7e-56ebf5f3f57a.png)

**Comment by magnus-madsen** [comment]
We should probably also fix the lack of effect support.

**Comment by mlutze** [comment]
I think this should be taken care of by the pretty printing changes.

**Comment by magnus-madsen** [comment]
Do you mean something already merged or something planned?

(You can see the output with the --doc flag)

I think we were relying on this:

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala#L51

as an ad-hoc way to *mutably* propagate type variable *names*. But I think that approach is now broken, because the way we check the signature against the inferred type is now correct (but uses fresh variables in the check):

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/ast/Scheme.scala#L121

TLDR; the connection between type variables and the names given to them by the user is messy. Question is how to clean it up, so that we can get nice documentation back.

**Comment by mlutze** [comment]
Ah right, I see.

My yet-unpushed changes to pretty printing I believe would show arbitrary letters rather than those assigned.

Perhaps we can add an optional param to Type.freshTypeVar to provide a name.

**Comment by magnus-madsen** [comment]
While the general problem of how to export information to Flixdoc continues to exist, I believe this specific problem has been solved.
